src/SMU-Keithley_4200-SCS/main.py

```diff
@@ -372,7 +376,7 @@ def configure(self):
                 # Current Range
                 if "Limited" in self.current_range:
                     current_range = self.current_ranges[self.current_range]
-                    self.set_current_range_limited(self.card_name, current_range)  # low range current
+                    self.set_current_range_limited(self.card_name[-1], current_range)  # low range current
```

Comment:
I tried to figure out how and why this fixes issues, but it seems quite difficult to understand what self.card_name is or it seems to be very different for various scenarios (sometimes the channel, sometimes a concatenation with the channel, sometimes a split...). An explanation when card_name is first defined would be quite useful.

Reply:
added a description of self.card_name